File: php/connect_april.php
<?php

class ConnectApril {
    public static function recv_request() {
        $n = end($argv);
        $filePath = "./{$n}.input";

        try {
            $data = file_get_contents($filePath);
            return json_decode($data, true);
        } catch (Exception $e) {
            echo "Error reading file: {$filePath}\n";
            return null;
        }
    }

    public static function send_response($response) {
        $n = end($argv);
        $filePath = "./{$n}.output";

        try {
            file_put_contents($filePath, json_encode($response));
        } catch (Exception $e) {
            echo "Error writing file: {$filePath}\n";
        }
    }
}
?>

File: matlab/connect_april.m
classdef ConnectApril
    methods(Static)
        function requestData = recv_request()
            n = str2double(varargin{end});
            filePath = sprintf('.%d.input', n);

            try
                data = fileread(filePath);
                requestData = jsondecode(data);
            catch
                fprintf('Error reading input file\n');
                requestData = [];
            end
        end

        function send_response(response)
            n = str2double(varargin{end});
            filePath = sprintf('.%d.output', n);

            try
                jsonStr = jsonencode(response);
                fid = fopen(filePath, 'w');
                fwrite(fid, jsonStr);
                fclose(fid);
            catch
                fprintf('Error writing output file\n');
            end
        end
    end
end